sn/src/routing/core/chunk_store/errors.rs

```
@@ -0,0 +1,52 @@
+// Copyright 2021 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::messaging::data::Error as ErrorMessage;
+use crate::types::convert_dt_error_to_error_message;
+use std::io;
+use thiserror::Error;
+use xor_name::XorName;
+
+/// Specialisation of `std::Result` for dbs.
+pub(crate) type Result<T, E = Error> = std::result::Result<T, E>;
+
+#[derive(Error, Debug)]
+#[non_exhaustive]
+/// Chunk Store error variants.
+pub enum Error {
+    /// Chunk not found.
+    #[error("Chunk not found: {0:?}")]
+    ChunkNotFound(XorName),
+    /// Invalid filename
+    #[error("Invalid chunk filename")]
+    InvalidFilename,
+    /// I/O error.
+    #[error("I/O error: {0}")]
+    Io(#[from] io::Error),
+    /// NetworkData error.
+    #[error("Network data error:: {0}")]
+    NetworkData(#[from] crate::types::Error),
+    /// No filename found
+    #[error("Path contains no file name")]
+    NoFilename,
```

not sure why this is different from `ChunkNotFound(XorName)`

---

The `NoFilename` should not happen unless the function `filepath_to_address` was called with a path that contains no filename. 
This should never be happening in the current code, I just didn't want to swallow and hide potential errors (if that function is used somewhere else at some point).